webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,288 @@
+            this.init();
+        } else if (this.props.connected && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate() {
+        if (this.props.isLoaded && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props) {
+        this.setState({
+            isLoaded: true,
+            ticketDetails: data.ticketDetails,
+        });
+    }
+
+    fixVersionLabel(fixVersion: string) {
+        if (fixVersion) {
+            const fixVersionString = 'Fix Version :';
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        let ticketStatusClass = 'default-style';
+
+        switch (ticketStatus.toLowerCase()) {
+        case statusIndeterminate:
+            ticketStatusClass += ' ticket-status--indeterminate';
+            break;
+        case statusDone:
+            ticketStatusClass += ' ticket-status--done';
+            break;
+        default:
+            ticketStatusClass += ' ticket-status--default';
+            break;
+        }
+
+        return <span className={ticketStatusClass}>{ticketStatus}</span>;
+    }
+
+    labelList(labels: string[]) {
+        if (!labels.length) {
+            return null;
+        }
+
+        let totalString = 0;
+        let totalHide = 0;
+        const labelList = labels.map((label: string, key: number): JSX.Element => {
+            if (totalString < 3) {
+                totalString++;
+                return (
+                    <span
+                        key={key}
+                        className='popover-labels__label-list'
+                    >
+                        {label}
+                    </span>);
+            }
+
+            totalHide++;
+            if (key === labels.length - 1) {
+                const moreLabels = `+${totalHide}more`;
+                return (
+                    <span
+                        key={key}
+                        className='popover-labels__label-list'
+                    >
+                        {moreLabels}
+                    </span>);
+            }
+
+            return <></>;
+        });
+        return (<div className='popover-labels__label'>{labelList}</div>);
+    }
+
+    render() {
+        if (!this.state.isLoaded) {
+            return (<p/>);
+        }
+
+        const {ticketDetails, href: jiraTicketURI} = this.state;
+        const jiraAvatar = ticketDetails && ticketDetails.jiraIcon;
```

Comment:
```typescript
ticketDetails?.jiraIcon is equivalent
```

Reply:
why not creating a type with all these defaulting to null and then filling it if these properties are in the ticketDetails?

Reply:
> ```ts
> ticketDetails?.jiraIcon is equivalent
> ```

Optional chaining is not set up in this plugin. Created the below issue for the same:
https://github.com/mattermost/mattermost-plugin-jira/issues/889